src/connection.js

```diff
@@ -405,6 +422,62 @@ export default class Connection extends EventEmitter {
     return this.messageIo.on('secure', this.emit.bind(this, 'secure'));
   }
 
+  resolveIPaddresses(host, callback) {
+    let ipv4Addresses = undefined;
+    let ipv6Addresses = undefined;
+    //resolve IPV4
+    Dns.resolve4(host, (err, addresses) => {
```

Comment:
You need to handle `err` by passing it back to `callback` and add correct error handling to wherever `resolveIPaddresses` is called from.